### DISCUSSION TITLE
Is meter type for watermeter with liters correct

### DISCUSSION THREAD

**Discussion by VincentVerweij** [jomjol/AI-on-the-edge-device#1499] [Q&A] [answered]
First of all a big thank you to the contributors of this `AI-on-the-edge-device` repository, it is becoming a more mature tool. Especially the last addition of Home Assistant discovery is really cool.

I am currently running version v13.0.1 and no complains.

I do have a question, hence opening it here as I am not sure if it's a bug since it does work but maybe not how I expect it to work.

So, I have my watermeter's `Meter type` set to `Watermeter (Value: l, Rate: l/h)`.  
In Home Assistant I do get the figures in my Energy graph:
![image](https://user-images.githubusercontent.com/41574930/205865737-48de0671-c3e3-4f76-9c06-bddb8229933c.png)

The thing is that I would expect these numbers when selecting cubic meter and not liters. For liters I would expect to see 10 liters instead of 0,01L. It would be awesome if my water consumption is that low, but unfortunately it isn't 😄 

Is this something wrong in my configuration?

In Home Assistant on the watermeter device, I see these values:
![image](https://user-images.githubusercontent.com/41574930/205866572-d9d2cdd2-e661-4a5a-a53f-07d7e279c340.png)

The `Value` is showing my actual meter's value which is correct. But the `Change since last digitalization` seems a bit off.

So any suggestions?

Again, awesome work you guys do. I am a programmer myself and I would really love to contribute, but C/C++ is not my cup of tea (more of a higher level programmer). I would probably push the stability in the wrong direction with my commits 😅

**Comment by VincentVerweij** [comment, marked as answer]
I just saw something in Home Assistant that might explain why it is shown in such a weird way:
![image](https://user-images.githubusercontent.com/41574930/205869287-4f7cb144-e171-49ba-ae93-9348909f2545.png)

Is it something that has to be configured in Home Assistant as well? That it would read those values as liters instead of cubic meters or gallons. From that little description it seems that it is expecting me to send cubic meters or gallons.
So, probably changing my configuration to send cubic meters would show better numbers (not tested yet).

**Reply by caco3** [reply]
Thanks for your feedback!

Your issue should be solved if you select the watermeter with m3 in the config.

**Comment by VincentVerweij** [comment]
I tried so indeed and am seeing proper value in the water consumption.
Why was there the need for other values if this does show everything properly?
Just curious 😄

**Reply by caco3** [reply]
Since there are also other systems than Homeassistant, resp. Homeassistant also can be used with `gallons` instead `m3`,  I decided to add all useful options.
